go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -245,20 +247,32 @@ func runOvnKube(ctx *cli.Context) error {
 		// since we capture some metrics in Start()
 		metrics.RegisterMasterMetrics(ovnNBClient, ovnSBClient)
 
-		ovnController := ovn.NewOvnController(ovnClientset, factory, stopChan, nil, ovnNBClient, ovnSBClient, util.EventRecorder(ovnClientset.KubeClient))
+		ovnController := ovn.NewOvnController(ovnClientset, masterWatchFactory, stopChan, nil, ovnNBClient, ovnSBClient, util.EventRecorder(ovnClientset.KubeClient))
 		if err := ovnController.Start(ovnClientset.KubeClient, master, wg); err != nil {
 			return err
 		}
 	}
 
 	if node != "" {
+		var nodeWatchFactory factory.NodeWatchFactory
+		if masterWatchFactory == nil {
```

Comment:
I didn't even know you could run ovnkube in master *and* node mode. I thought it had to be either or.

Reply:
I'm not sure we test it, but it definitely was used.